frontend/simple/views/components/Chatroom/ChatMain.vue

```diff
@@ -0,0 +1,390 @@
+    }
+  }
+
+  .c-footer {
+    position: relative;
+    padding-left: $gi-spacer;
+    padding-right: $gi-spacer;
+  }
+}
+
+</style>
+<script>
+import MainHeader from '../MainHeader.vue'
+import Avatar from '../Avatar.vue'
+import Loading from '../Loading.vue'
+import { List } from '../Lists/index.js'
+import { MenuParent, MenuTrigger, MenuContent, MenuHeader, MenuItem } from '../Menu/index.js'
+import Message from './Message.vue'
+import MessageInteractive from './MessageInteractive.vue'
+import MessageNotification from './MessageNotification.vue'
+import ConversationGreetings from '../../containers/Chatroom/ConversationGreetings.vue'
+import SendArea from './SendArea.vue'
+import { currentUserId } from '../../containers/Chatroom/fakeStore.js'
+
+export default {
+  name: 'ChatMain',
+  components: {
+    MainHeader,
+    Avatar,
+    Loading,
+    List,
+    MenuParent,
+    MenuTrigger,
+    MenuContent,
+    MenuHeader,
+    MenuItem,
+    Message,
+    MessageInteractive,
+    MessageNotification,
+    ConversationGreetings,
+    SendArea
+  },
+  props: {
+    summary: {
+      type: Object, // { type: String, title: String, description: String, routerBack: String }
+      default: {}
+    },
+    details: {
+      type: Object, // { isLoading: Bool, conversation: Array, participants: Object }
+      default: {}
+    }
+  },
+  data () {
+    return {
+      config: {
+        isPhone: null
+      },
+      ephemeral: {
+        bodyPaddingBottom: '',
+        conversationIsLoading: false,
+        pendingMessages: []
+      }
+    }
+  },
+  created () {
+    // TODO #492 create a global Vue Responsive just for media queries.
+    var mediaIsPhone = window.matchMedia('screen and (max-width: 639px)')
+    this.config.isPhone = mediaIsPhone.matches
+    mediaIsPhone.onchange = (e) => { this.config.isPhone = e.matches }
+  },
+  updated () {
+    if (this.summary.title) {
+      // force conversation viewport to be at the bottom (most recent messages)
+      if (this.config.isPhone) {
+        setTimeout(function () {
+          window.scroll(0, document.body.offsetHeight)
+        }, 0) // REVIEW: without timeout it scrolls before the DOM is ready... why?
+      } else {
+        this.$refs.conversation && this.$refs.conversation.scroll(0, this.$refs.conversation.scrollHeight)
+      }
+    }
+  },
+  computed: {
+    bodyStyles () {
+      return this.config.isPhone ? { paddingBottom: this.ephemeral.bodyPaddingBottom } : {}
+    },
+    startedUnreadIndex () {
+      return this.details.conversation.findIndex(message => message.unread === true)
+    },
+    getMessageAt () {
+      let isCurrentUser
+
+      return this.details.conversation.map((message, index) => {
+        isCurrentUser = this.currentUserAttr.id === message.from
+
+        if (message.from === 'interactive') {
+          return {
+            is: MessageInteractive,
+            id: message.id
+          }
+        }
+        if (message.from === 'notification') {
```

Comment:
It is considered "code smell" to do raw string constant comparisons like this, because it's not clear where these strings come from, and if they need to be modified they have to be modified in a bunch of different places.

So it is best to store them some place and then reference them as constants, similar to how you created a file with constants for all the events in the app in `util/events.js`. For these you don't necessarily need to create a new file (but you can), they could come from one of the components somehow, or whatever.

Reply:
Totally agreed, I've forgot these ones. Since these "types" are comparing something that comes from the store, I added it to `fakeStore.js` for now as an object `messageTypes`